src/Spryker/Zed/EventBehavior/Business/EventBehaviorFacadeInterface.php

```diff
@@ -94,6 +94,20 @@ public function executeResolvedPluginsBySources(array $resources, array $ids = [
      */
     public function getAvailableResourceNames(): array;
 
+    /**
+     *
+     * Specification:
+     * - Returns an array of foreign keys grouped by column.
+     *
+     * @api
+     *
+     * @param \Generated\Shared\Transfer\EventEntityTransfer[] $eventTransfers
+     * @param string $foreignKeyColumnName
+     *
+     * @return array ['foreignKey' => ['relatedForeignKeyName1' => relatedForeignKeyValue1, ...]]
+     */
+    public function getGroupedEventTransferForeignKeysByColumn(array $eventTransfers, string $foreignKeyColumnName): array;
```

Comment:
@ehsanmx Please verify if the facade naming fits

Reply:
I would go for `getEventTransfersGroupedByForeignKeys`

Reply:
Please also move this method close to other `getEventTransfers....`

Reply:
we changed it to `getGroupedEventTransferForeignKeysByForeginKey`